colibri/src/icons.rs

```diff
@@ -119,6 +112,96 @@ async fn query_token_icon_and_extension(
     None
 }
 
+async fn query_uniswap_v3_position_icon(
+    chain_id: u64,
+    token_id: &str,
+    contract_address: Address,
+    inquirer: Arc<EvmNodeInquirer>,
+) -> Option<(Bytes, &'static str)> {
+    let token_id: U256 = token_id.parse()
+        .map_err(|e| {
+            error!("Invalid token ID '{}' for NFT position on chain ID {} ({}): {}", 
+            token_id, chain_id, inquirer.blockchain.as_str(), e)
+        })
+        .ok()?;
+
+    for node in inquirer.rpc_nodes.read().await.clone() {
+        let provider = match inquirer.connect_to_node(&node).await {
```

Comment:
So wait ... you have to redo th econnection to a node each time you want to query an icon? Shouldnt' connections be cached somewhere?